cypress/integration/composer_selection_spec.js

```diff
@@ -0,0 +1,369 @@
+  publishButton().should('be.visible');
+  publishButton().should('be.disabled');
+};
+
+const publishButtonIsEnabled = () => {
+  publishButton().should('be.visible');
+  publishButton().should('not.be.disabled');
+};
+
+const communityDropdown = () =>
+  cy.get('[data-cy="composer-community-selector"]');
+const channelDropdown = () => cy.get('[data-cy="composer-channel-selector"]');
+const communitySelected = () =>
+  cy.get('[data-cy="composer-community-selected"]');
+const channelSelected = () => cy.get('[data-cy="composer-channel-selected"]');
+const inboxComposeButton = () => cy.get('[data-cy="inbox-view-post-button"]');
+const everythingFilter = () =>
+  cy.get('[data-cy="inbox-community-list-item"]').first();
+const firstCommunityFilter = () =>
+  cy.get('[data-cy="inbox-community-list-item"]').eq(1);
+const secondCommunityFilter = () =>
+  cy.get('[data-cy="inbox-community-list-item"]').eq(2);
+const inboxChannelFilter = () =>
+  cy.get('[data-cy="inbox-channel-list-item"]').first();
+const secondInboxChannelFilter = () =>
+  cy.get('[data-cy="inbox-channel-list-item"]').eq(1);
+const composerPlaceholder = () =>
+  cy.get('[data-cy="thread-composer-placeholder"]');
+const titlebarComposeButton = () =>
+  cy.get('[data-cy="titlebar-compose-button"]');
+
+const communityDropdownIsEnabled = () => {
+  communityDropdown().should('be.visible');
+  communityDropdown().contains('Choose a community');
+  communityDropdown().contains('Spectrum');
+  communityDropdown().contains('Payments');
+};
+
+const channelDropdownIsHidden = () => {
+  channelDropdown().should('not.be.visible');
+};
+
+const channelDropdownIsEnabled = () => {
+  channelDropdown().should('be.visible');
+  channelDropdown().contains('Choose a channel');
+};
+
+const communityIsLocked = () => {
+  communitySelected().should('be.visible');
+};
+
+const channelIsLocked = () => {
+  channelSelected().should('be.visible');
+};
+
+const openMobileCommunityMenu = () => {
+  cy.get('[data-cy="community-menu-open"]')
+    .first()
+    .click();
+};
+const closeMobileCommunityMenu = () => {
+  cy.get('[data-cy="community-menu-close"]')
+    .first()
+    .click('right', { force: true });
+};
+
+describe('/new/thread community and channel selection', () => {
+  beforeEach(() => {
+    cy.auth(user.id);
+  });
+
+  it('does not select a community or channel if no query params are passed', () => {
+    cy.visit('/new/thread');
+    publishButtonIsDisabled();
+    communityDropdownIsEnabled();
+    channelDropdownIsHidden();
+  });
+
+  it('fetches channels when a community is selected', () => {
+    cy.visit('/new/thread');
+    communityDropdownIsEnabled();
+    channelDropdownIsHidden();
+
+    communityDropdown().select('Spectrum');
+    channelDropdownIsEnabled();
+    channelDropdown().contains('General');
+    channelDropdown().contains('Private');
+
+    communityDropdown().select('Payments');
+    channelDropdown().contains('General');
+    channelDropdown().contains('Private');
```

Comment:
We should add a third channel to Payments to make sure it fetched the new channels